src/mavedb/lib/permissions.py

```diff
@@ -178,8 +178,11 @@ def has_permission(user_data: Optional[UserData], item: Base, action: Action) ->
             else:
                 return PermissionResponse(False)
         elif action == Action.ADD_SCORE_SET:
+            # Not allow add score set in meta-analysis experiments.
+            if any(score_set.meta_analyzes_score_sets for score_set in item.score_sets):
+                return PermissionResponse(False, 403, "not allow to add score set in meta-analysis experiment")
             # Only permitted users can add a score set to a private experiment.
-            if user_may_edit or roles_permitted(active_roles, [UserRole.admin]):
+            elif user_may_edit or roles_permitted(active_roles, [UserRole.admin]):
```

Comment:
Instead of this being a check in the permissions function, I think it would be better if we check this right before adding the score set. I guess I would argue the user does technically have permission to add a score set to an experiment, it is just that in this particular case we don't allow that for these particular score set types.

Reply:
I struggled with where to put these checking codes before. I moved it to the `create_score_set` function. I think it's better than permissions function. The final effects are the same on the MaveDB website.